### PR TITLE
Update and refine settings descriptions

### LINKED ISSUE #956: Update settings.md to include all the latest settings changes

### PR DESCRIPTION
**Pull request by plemarquand**
- Remove redundant settings descriptions from settings.md, moving any extra information found there in to the setting description shown in the extension settings page.
- Add information on how to enable experimental sourcekit-lsp features to settings.md.
- Consistently apply markdown formatting to setting descriptions.

Fixes #956

### CODE REVIEW

**Comment by plemarquand** [comment]
@adam-fowler Instead of maintaining two largely identical settings descriptions I just rolled everything in to the settings descriptions in the package.json. I think this should be the single source of truth as most users won't seek out the settings.md page, and the'll see these settings descriptions right in the app. If we really want to do a deep dive into a specific setting, so much so that its too much information to have in the extension settings page, we can link out to the settings.md document and expand on it there, but there aren't any that are that long today.

settings.md does now contain some settings combinations/configurations that can be used to accomplish specific things. For instance right now it has sourcekit-lsp server arguments you can set to enable background indexing or peek macro.

File: package.json [hunk 1]
```diff
@@ -258,14 +258,14 @@
...
-            "markdownDescription": "Disable automatic running of `swift package resolve`. This will also disable searching for command plugins and the initial test discovery process as both of these can kick of a resolve.",
+            "markdownDescription": "Disable automatic running of `swift package resolve` whenever the `Package.swift` or `Package.resolve` files are updated. This will also disable searching for command plugins and the initial test discovery process.",
             "scope": "machine-overridable",
             "order": 7
           },
           "swift.diagnosticsCollection": {
             "type": "string",
             "default": "keepSourceKit",
-            "description": "Controls how diagnostics from the various providers are merged into the `swift` diagnostics collection.",
+            "markdownDescription": "Controls how diagnostics from the various providers are merged into the `swift` diagnostics collection.",
```
Comment (adam-fowler) [inline package.json:268]: Should probably make it clear this is related to diagnostics used in the problems pane. The word diagnostic is not used in the UI anywhere

File: package.json [hunk 2]
```diff
@@ -285,23 +285,23 @@
           "swift.diagnosticsStyle": {
             "type": "string",
             "default": "llvm",
-            "description": "Controls which -diagnostic-style option to pass to `swiftc` when running `swift` tasks.",
+            "markdownDescription": "Controls which `-diagnostic-style` option to pass to `swiftc` when running `swift` tasks.",
```
Comment (adam-fowler) [inline package.json:288]: Indicate what effect this has?

File: package.json [hunk 3]
```diff
@@ -334,14 +334,14 @@
...
-            "description": "Controls whether or not the extension will warn about being unable to create symlinks. (Windows only)",
+            "markdownDescription": "Controls whether or not the extension will warn about being unable to create symlinks. (Windows only)",
             "scope": "application",
             "order": 14
           },
           "swift.enableTerminalEnvironment": {
             "type": "boolean",
             "default": true,
-            "description": "Controls whether or not the extension will contribute Swift environment variables to the integrated terminal.",
+            "markdownDescription": "Controls whether or not the extension will contribute Swift environment variables to the integrated terminal.",
```
Comment (adam-fowler) [inline package.json:344]: Indicate which settings might add environment variables

File: package.json [hunk 4]
```diff
@@ -525,13 +525,13 @@
           "swift.debugger.useDebugAdapterFromToolchain": {
             "type": "boolean",
             "default": false,
-            "description": "Use lldb debug adapter packaged with Swift toolchain as your debug adapter. This is currently only available on Windows platforms or macOS Swift 6",
+            "markdownDescription": "Use lldb debug adapter packaged with Swift toolchain as your debug adapter. This is currently only available on Windows platforms or macOS Swift 6",
```
Comment (adam-fowler) [inline package.json:528]: We can remove the `macOS` from here as the DAP is available on Linux as well

**Review by adam-fowler — CHANGES_REQUESTED** [review]
Combining the two does make it easier to keep everything up to date
I have a few requests for changes.

**Comment by plemarquand** [comment]
@swift-server-bot test this please

File: package.json [hunk 5]
```diff
@@ -525,13 +525,13 @@
           "swift.debugger.useDebugAdapterFromToolchain": {
             "type": "boolean",
             "default": false,
-            "description": "Use lldb debug adapter packaged with Swift toolchain as your debug adapter. This is currently only available on Windows platforms or macOS Swift 6",
+            "markdownDescription": "Use lldb debug adapter packaged with Swift toolchain as your debug adapter. This is currently only available on Windows platforms using Swift 6",
```
Comment (adam-fowler) [inline package.json:528]: I didn't mean remove the `or`. This works for any platform running Swift 6 now
```suggestion
            "markdownDescription": "Use lldb debug adapter packaged with Swift toolchain as your debug adapter. This is currently only available on Windows or platforms using Swift 6",
```

**Review by adam-fowler — APPROVED** [review]
I made one minor change as lldb-dap is available on all platforms using Swift 6
Everything else looks good

**Pull request merged**